### PR TITLE
Port json tests

### PR DESCRIPTION
**Pull request by srikrsna-buf**
Port json tests from v1. Covers most of the cases. Need to add tests for groups and merge.

### CODE REVIEW

File: packages/protobuf-test/src/next/json.test.ts
Comment (timostamm) [inline packages/protobuf-test/src/next/json.test.ts:333]: Sounds good to me to add JSON tests for WKT here, and keep other WKT-specific tests in `wkt` (v1 has them in `google/protobuf`).

File: packages/protobuf-test/src/next/json.test.ts
Comment (timostamm) [inline packages/protobuf-test/src/next/json.test.ts:353]: This works well and looks good. But ion general, let's be cautious adding functions that create tests, and prefer explicit `describe` / `test` for consistency and readability.

File: packages/protobuf-test/src/next/json.test.ts
Comment (timostamm) [inline packages/protobuf-test/src/next/json.test.ts:272]: LGTM after merging in b721c26 and uncommenting this.

File: packages/protobuf-test/src/next/json.test.ts
Comment (srikrsna-buf) [reply packages/protobuf-test/src/next/json.test.ts]: Removed the test call and renamed the function.

Comment (timostamm) [reply packages/protobuf-test/src/next/json.test.ts]: Unfortunately, rolling your own assertion functions with Jest has problems. See here:

```
Summary of all failing tests
 FAIL  next/json.test.js
  ● Test suite failed to run

    expect(received).toStrictEqual(expected) // deep equality

    Expected: true
    Received: {"boolValueField": true, "bytesValueField": "aGVsbG8gd29ybGQ=", "doubleValueField": 1.2, "floatValueField": 1.3, "int32ValueField": 6, "int64ValueField": "4", "stringValueField": "a", "uint32ValueField": 7, "uint64ValueField": "5"}

      462 | function assertJson(desc, init, json) {
      463 |     const msg = create(desc, init);
    > 464 |     expect(toJson(desc, msg)).toStrictEqual(true);
          |                               ^
      465 |     expect(fromJson(desc, json)).toStrictEqual(msg);
      466 | }
      467 |

      at assertJson (next/json.test.js:464:31)
```

What we want to see is the test "json serialization > scalar fields" failing, but instead the test suite errors, and does not show which test failed, and since line numbers point to the transpiled .js files, this becomes difficult to debug.

I _think_ that the previous version that created the test did not have this issue. If it doesn't it's better to revert.

Issues like this are the reason some existing tests are so verbose. V1 has `testMT()`, but I wish it didn't. It's best to stick with what Jest suggests in their docs, avoid test helpers, and use variables scoped to `describe`, with `beforeEach`, if necessary.

Can you revise?

Comment (srikrsna-buf) [reply packages/protobuf-test/src/next/json.test.ts:582]: Done

**Review by timostamm — APPROVED** [review]

**Pull request merged**
